### PR TITLE
Add flag to configure whether the same day of every year should be drawn as selected.(Default is true as is.)

### PR DESCRIPTION
**Pull request by scm573**
#170 

All tests passed.

### CODE REVIEW

File: library/src/main/java/com/github/sundeepk/compactcalendarview/CompactCalendarController.java [hunk 1]
```diff
@@ -78,6 +78,7 @@
     private boolean shouldDrawIndicatorsBelowSelectedDays = false;
     private boolean displayOtherMonthDays = false;
     private boolean shouldSelectFirstDayOfMonthOnScroll = true;
+    private boolean shouldSelectSameDayOfEveryYear = true;
```
Comment (scm573) [inline library/src/main/java/com/github/sundeepk/compactcalendarview/CompactCalendarController.java]: Perhaps "shouldDrawSameDayOfEveryYear" is better?

**Comment by SundeepK** [comment]
I should update the readme and gradle file to execute additional tests. But there are also screenshot tests:
https://github.com/facebook/screenshot-tests-for-android
http://facebook.github.io/screenshot-tests-for-android/#getting-started

You will need to configure your system to execute the tests. Then from command line, root of the project:

```./gradlew verifyMode screenshotTests```

This will run the screenshot tests to ensure nothing it broken in the sample app provided. 

You can add a new test in here:

https://github.com/SundeepK/CompactCalendarView/blob/master/sample/src/androidTest/java/sundeepk/github/com/sample/ApplicationTest.java

It uses the sample app to verify behavoiur is correct. Alot of tests will take a screenshot. When you run the verifyMode task, it will execute all the tests and compare them to already existing screenshots to make sure nothing has changed/broken.

So you can add a tests which scrolls 12 months in the future and take a screenshots and then run 
`./gradlew recordMode screenshotTests` to generate an expected screenshot first time round. Check that screenshot in so that when we execute all the screenshot tests, we know that this feature is covered.

You can use the method `scrollCalendarForwardBy(12)` to scroll the calendar by 12 months in yuor test.

**Comment by scm573** [comment]
Hi, thanks for the reply.

I'll try to add new UI test cases. But one thing I'm wondering is, those screenshot tests seems depending on the tested device. I tried the gradle task, the tests passed but it failed when comparing the screenshots. And they're indeed different as I check.
So maybe I need the information of the device you tested on?

**Comment by SundeepK** [comment]
Ah thanks for trying it out. Maybe it's time I integrated some CI for the library that can run the UI tests automatically. 

I used a nexus 6p device to generate and test using the screenshot library. What changes did you see? Maybe it might work with an emulator but I have not tried it yet. Worst case, I can add the UI test and run the tests locally to unblock the pull request.

**Comment by scm573** [comment]
Hmm... I tried again on Nexus 6P emulator with API23, I got the most similar screenshots so far (I think they should be same by my eyes) , but unfortunately they still lead to android_screenshot_tests.recorder.VerifyError.
And actually there're a little differences between the origin screenshots and those generated from my tests: the origin images have a smaller file size, and no "Dimensions" property shown in Finder. (I'm using Mac)

p.s. I guess your test device's OS is 6.0? Seems that one or two UI test cases are unstable on 5 or 7.

**Comment by SundeepK** [comment]
I'm running version 7.1.1 on an actual device. Perhaps the best thing todo is to always run the tests on an emulator and save images generated from the emulator instead of a real device. It might be more reproducable since everyone should have access to emulator and might be better when intergrating with some CI chain in the future.

Maybe the best thing todo is to park this for now. If you have created the test then just check it into the branch and I can verify things locally when I get the time, whilst I think of a more automated way of doing it. Also this test can be done in unit tests but I much prefer the screenshot tests. Unfortunatly, they are not as robust it seems :/

**Comment by scm573** [comment]
Hi,

I tried again on emulators of each API from 5.0 to 7.1. Sadly the assertion were all failed in this case:
[https://github.com/SundeepK/CompactCalendarView/blob/master/sample/src/androidTest/java/sundeepk/github/com/sample/ApplicationTest.java#L171](url)
Perhaps the scrolling method doesn't work well in different environments.

As well as the screenshots, I think it seems better to fist bump to you, so that we could make those tests unified.
I'm looking forward to your updates, and glad to be of help.

**Comment by SundeepK** [comment]
Just an update, I was away for a a few days so I didn't have any chance to have a further look at the pull request. I expect to have time this weekend so expect some update soon.

Thanks.

**Comment by SundeepK** [comment]
So after having a good look through and thinking about things further. I think we should just make this the default functionality and get rid the boolean value to configure things. I doubt any one who makes use of the `compactCalendarShouldSelectFirstDayOfMonthOnScroll` to false will miss that functionality. Plus it just makes things easier to understand.

If people complain then we can add that attribute to make it a configurable value I think. 

So:
`compactCalendarShouldSelectFirstDayOfMonthOnScroll` set to false will mean only the current selected day is selected and no other year will be highlighted.

Setting `compactCalendarShouldSelectFirstDayOfMonthOnScroll` to true or default value, will continue to select the first day on scroll and any years ahead/behind.

I'll checkout your changes and just get rid of the extra `compactCalendarShouldSelectSameDayOfEveryYear` xml attribute so to keep your commit history.

**Comment by SundeepK** [comment]
Released in:
```
dependencies {
    compile 'com.github.sundeepk:compact-calendar-view:2.0.1'
}
```

**Pull request merged**